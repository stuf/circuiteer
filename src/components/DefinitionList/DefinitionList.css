.definition-list {
  @apply grid gap-2;
  grid-template-columns: 1fr 4fr;
}

.definition-list__key {
  @apply font-bold;
}

.definition-list__value {
  @apply font-mono;
}
