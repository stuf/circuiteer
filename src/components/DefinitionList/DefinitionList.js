import { Fragment } from 'react';
import * as P from 'prop-types';
import cx from 'classnames';

import './DefinitionList.css';

export function DefinitionList(props) {
  const { className, items, name } = props;

  const cns = {
    root: ['definition-list', className],
    key: ['definition-list__key'],
    value: ['definition-list__value'],
  };

  return (
    <dl className={cx(cns.root)}>
      {items.map(({ id, key, value }, ix) => {
        const kvName = [name, id].join('-');

        return (
          <Fragment key={ix}>
            <dt className={cx(cns.key)} id={kvName}>
              {key}
            </dt>
            <dd className={cx(cns.value)} aria-labelledby={kvName}>
              {value}
            </dd>
          </Fragment>
        );
      })}
    </dl>
  );
}

DefinitionList.propTypes = {
  className: P.string,
  items: P.arrayOf(
    P.shape({
      id: P.string,
      key: P.string,
      value: P.oneOfType([P.node, P.elementType, P.string]),
    }),
  ).isRequired,
  name: P.string.isRequired,
};

DefinitionList.defaultProps = {
  items: [],
};
