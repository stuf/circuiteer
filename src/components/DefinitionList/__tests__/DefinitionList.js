import { render, logDOM, logRoles } from '@testing-library/react';
import { DefinitionList } from '../DefinitionList';

describe('DefinitionList', () => {
  test('works', async () => {
    const name = 'foo';
    const items = [
      { id: 'one', key: 'One', value: '1' },
      { id: 'two', key: 'Two', value: '2' },
    ];
    const { container, findByRole } = render(
      <DefinitionList {...{ name, items }} />,
    );

    await findByRole('definition', { name: /one/i });

    expect(container).toMatchSnapshot();
  });
});
