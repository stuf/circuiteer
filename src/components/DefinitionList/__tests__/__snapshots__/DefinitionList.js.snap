// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefinitionList works 1`] = `
<div>
  <dl
    class="definition-list"
  >
    <dt
      class="definition-list__key"
      id="foo-one"
    >
      One
    </dt>
    <dd
      aria-labelledby="foo-one"
      class="definition-list__value"
    >
      1
    </dd>
    <dt
      class="definition-list__key"
      id="foo-two"
    >
      Two
    </dt>
    <dd
      aria-labelledby="foo-two"
      class="definition-list__value"
    >
      2
    </dd>
  </dl>
</div>
`;
